ext/flexmailer/src/Validator.php

```diff
@@ -25,14 +25,20 @@ class Validator {
   const EVENT_CHECK_SENDABLE = 'civi.flexmailer.checkSendable';
 
   /**
-   * @param \CRM_Mailing_DAO_Mailing $mailing
+   * @param array $params
    *   The mailing which may or may not be sendable.
    * @return array
    *   List of error messages.
    */
-  public static function createAndRun($mailing) {
-    $validator = new \Civi\FlexMailer\Validator();
-    return $validator->run(array(
+  public static function createAndRun(array $params): array {
```

Comment:
@colemanw @eileenmcnaughton - The failures in `ValidatorTest` are complaining because the signature changed (DAO => array).

It should be straight-forward to update the test:

```diff

diff --git a/ext/flexmailer/tests/phpunit/Civi/FlexMailer/ValidatorTest.php b/ext/flexmailer/tests/phpunit/Civi/FlexMailer/ValidatorTest.php
index 35541e5e67..642d4527b2 100644
--- a/ext/flexmailer/tests/phpunit/Civi/FlexMailer/ValidatorTest.php
+++ b/ext/flexmailer/tests/phpunit/Civi/FlexMailer/ValidatorTest.php
@@ -86,9 +86,7 @@ class ValidatorTest extends \CiviUnitTestCase {
    * @dataProvider getExamples
    */
   public function testExamples($mailingData, $expectedErrors): void {
-    $mailing = new \CRM_Mailing_DAO_Mailing();
-    $mailing->copyValues($mailingData);
-    $actualErrors = Validator::createAndRun($mailing);
+    $actualErrors = Validator::createAndRun($mailingData);
     $this->assertEquals(
       array_keys($actualErrors),
       array_keys($expectedErrors)
```

Grepped `universe` for `createAndRun` -- I don't see anything outside of `civicrm-core` that uses it.